### PR TITLE
Mills Provosty

### PR DESCRIPTION
**Pull request by MillsProvosty**
This was so fun!
So, I could see that in that last method we'd need to take the subtract the pictures date from the artists birth date, set that to the key and transform the value to the picture name. But, the artist being passed in was nil....so, I was back at find artist by ID thinking of doing an if statement of if it's nil return the name?????? Not sure. Anyway, I completed iteration 3, and got the first three done for iteration 4.

### CODE REVIEW

File: lib/artist.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  def initialize(attributes)
+    @id      = attributes[:id]
+    @name    = attributes[:name]
+    @born    = attributes[:born]
+    @died    = attributes[:died]
+    @country = attributes[:country]
+  end
+
+end
```
Comment (mstang15) [inline lib/artist.rb]: 💯

Comment (MillsProvosty) [reply lib/artist.rb]: Also, I figured out what the problem was- I didn't put in the line assigning artist id to the instance variable of Diane Whatshername. Had I done that, I could have gotten further on that last method....not sure if I'd have finished it, but, that was the nil problem.

File: lib/curator.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    end
+
+    artist = []
+    photo_by_artist.find do |id|
+      if photo_by_artist.count(id) > 1
+         artist << find_artist_by_id(id)
+      end
+    end
+    artist
+  end
```
Comment (mstang15) [inline lib/curator.rb]: This is super interesting, I haven't seen a solution like this before... nice problem solving.

File: lib/curator.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    end
+    @photographs
+  end
+
+  def load_artists(location)
+    CSV.foreach(location, {:headers => true, :header_converters => :symbol}) do |row|
+      @artists << Artist.new(row)
+    end
+    @artists
+  end
```
Comment (mstang15) [inline lib/curator.rb]: I really like the way you're reading the rows and creating new artist and photo objects. Nice job!

File: lib/curator.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  def photographs_taken_between(range)
+    @photographs.find_all do |photo|
+      range.cover?(photo.year.to_i)
+    end
+  end
```
Comment (mstang15) [inline lib/curator.rb]: ooooh fun .. haven't seen .cover? used either!

File: lib/photograph.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+              :artist_id,
+              :year
+
+  def initialize(attributes)
+    @id        = attributes[:id]
+    @name      = attributes[:name]
+    @artist_id = attributes[:artist_id]
+    @year       = attributes[:year]
+  end
+end
```
Comment (mstang15) [inline lib/photograph.rb]: 💯

**Pull request closed without merging**